src/transformers/models/llama/modeling_llama.py

```diff
@@ -464,8 +437,15 @@ def forward(
             key_states = key_states.to(target_dtype)
             value_states = value_states.to(target_dtype)
 
-        attn_output = self._flash_attention_forward(
-            query_states, key_states, value_states, attention_mask, q_len, dropout=dropout_rate
+        attn_output = _flash_attention_forward(
+            query_states,
+            key_states,
+            value_states,
+            attention_mask,
+            q_len,
+            dropout=dropout_rate,
+            sliding_window=getattr(self, "sliding_window", None),
+            cache_position=cache_position,
```

Comment:
`is_causal`, `_flash_attn_uses_top_left_mask` are not passed although they should